hydra/hydra.py

```diff
@@ -280,24 +297,60 @@ def put_owner(self, case_number, user=""):
-
-       if status: query_params.update({'status': ", ".join(status)})
-       if fields: query_params.update({'fields': ", ".join(fields)})
-       if accounts: query_params.update({'accounts': ", ".join(accounts)})
-       if cases: query_params.update({'cases': ", ".join(cases)})
-       if sbrGroups: query_params.update({'sbrGroups': ", ".join(sbrGroups)})
-       if needsNewOwner: query_params.update({'needsNewOwner': needsNewOwner})
-       if severity: query_params.update({'severity': ", ".join(severity)})
-       if serviceLevel:
-           query_params.update({'serviceLevel': ", ".join(serviceLevel)})
-       if fts: query_params.update({'fts': fts})
-       if ownerSsousername:
-           query_params.update({'ownerSsousername': ", ".join(ownerSsousername)})
-       if tags:
-           query_params.update({'tags': ", ".join(tags)})
-
-       return self.__get_api('cases/', parameters=query_params)
+    def query_cases(self, fields=[], accounts=[], cases=[],
+           sbrGroups=[], ownerSsousername=[], tags=[], cluster_ids=[]):
+
+        query_params = {}
+        ## Pulls back a lot if your not filtering! A field filter is recommended.
+        if fields: query_params.update({'fl': ",".join(fields)})
+
+        ## Mutually Exclusive Search!!!
+        ## You can only search based on 1 paramiter! accounts, cases, etc.
+
+        if accounts:
+            if len(accounts) == 1:
+                query_params.update({'q':
+                    'case_accountNumber:{}'.format(accounts[0])})
+            else:
+                query_params.update({'q':
+                    'case_accountNumber:({})'.format(' OR '.join(accounts))})
+
+        if cases:
+            if len(cases) == 1:
+                query_params.update({'q': 'case_number:{}'.format(cases[0])})
+            else:
+                query_params.update({'q':
+                    'case_number:({})'.format(' OR '.join(cases))})
+
+        if ownerSsousername:
+            if len(ownerSsousername) == 1:
+                query_params.update({'q':
+                    'case_owner_sso_username:{}'.format(ownerSsousername[0])})
+            else:
+                query_params.update({'q':
+                    'case_owner_sso_username:({})'.format(' OR '.join(ownerSsousername))})
+
+        if sbrGroups:
+            if len(sbrGroups) == 1:
+                query_params.update({'q': 'case_sbr:{}'.format(sbrGroups[0])})
+            else:
+                query_params.update({'q':
+                    'case_sbr:({})'.format(' OR '.join(sbrGroups))})
+
+        if tags:
+            if len(tags) == 1:
+                query_params.update({'q': 'case_tags:{}'.format(tags[0])})
+            else:
+                query_params.update({'q':
+                    'case_tags:({})'.format(' OR '.join(tags))})
+
+        if cluster_ids:
+            if len(cluster_ids) == 1:
+                query_params.update({'q':
+                    'case_openshift_cluster_id:{}'.format(cluster_ids[0])})
```

Comment:
No need for this case, you can just `join` for any len >= 1:

```
>>> '-'.join(['a'])
'a'
>>> '-'.join(['a', 'b'])
'a-b'
```

Reply:
Fixed.